### ISSUE TITLE
queries executed in rails initializers and queries executed outside of request scope

### ISSUE THREAD

**Issue by dominh** [iMacTia/lograge-sql#24]
I've found a minor issue:

In my project, I'm using `lograge-sql`, and I perform some DB queries in initializers. However, these queries log the error:
```
Could not log "sql.active_record" event. NoMethodError: undefined method `call' for nil:NilClass ["/home/dominik/.rvm/gems/ruby-2.7.1@dataflow/gems/lograge-sql-1.1.0/lib/lograge/sql/extension.rb:42:in `sql'"
```

Looking into `lograge-sql` code, I spotted that `lograge-sql` is set up in `after_initialize` hook. So my dirty fix for this was to put the code below in the beginning of initializers:
```
Lograge::Sql.setup(config.lograge_sql)
```
This way I surpressed the error. However, SQL queries executed out of a request scope(Rails console, Initializers) are still not logged.

My initial idea to fix it was to move the code below into the `setup` method, and change it to actually not unsubscribe `ActiveRecord::LogSubscriber` but decorate it, but I'm not experienced with Lograge code, so I'm not going to develop it now
```
ActiveSupport::LogSubscriber.log_subscribers.each do |subscriber|
  Lograge.unsubscribe(:active_record, subscriber) if subscriber.is_a?(ActiveRecord::LogSubscriber)
end
```

**Comment by iMacTia** [comment]
Hello @dominh and thanks for raising this issue.
I *think* this might have been solved with #23, as we moved from `after_initialize` to `on_load`, however that has not been released yet.
Is there a chance you could point your project to the master branch of this gem and give it a go?
You can do so in the `Gemfile`, please let me know if you don't know how to do it.

**Comment by dominh** [comment]
Hello @iMacTia 
`to_prepare` hook is executed before `after_initialize` hook however is still executed after initializers. 
However, I think it's OK if we move 
```
Lograge::ActiveRecordLogSubscriber.attach_to :active_record
```
Into `setup` method(https://github.com/iMacTia/lograge-sql/pull/25). 
This way lograge won't bother for any request during initialization process.

Still `ActiveRecord::LogSubscriber` is completely removed in `setup`, which makes impossible to log queries from i.e Rails console or app-exit hooks
